homeassistant/components/control4/light.py

```diff
@@ -0,0 +1,219 @@
+        update_method=async_update_data_non_dimmer,
+        update_interval=timedelta(seconds=scan_interval),
+    )
+    dimmer_coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="light",
+        update_method=async_update_data_dimmer,
+        update_interval=timedelta(seconds=scan_interval),
+    )
+
+    # Fetch initial data so we have data when entities subscribe
+    await non_dimmer_coordinator.async_refresh()
+    await dimmer_coordinator.async_refresh()
+
+    items_of_category = await get_items_of_category(hass, entry, CONTROL4_CATEGORY)
+    for item in items_of_category:
+        if item["type"] == 7:
+            item_name = item["name"]
+            item_id = item["id"]
+            item_parent_id = item["parentId"]
+            item_is_dimmer = item["capabilities"]["dimmer"]
+
+            if item_is_dimmer:
+                item_coordinator = dimmer_coordinator
+            else:
+                item_coordinator = non_dimmer_coordinator
+
+            for parent_item in items_of_category:
+                if parent_item["id"] == item_parent_id:
+                    item_manufacturer = parent_item["manufacturer"]
+                    item_device_name = parent_item["name"]
+                    item_model = parent_item["model"]
+            async_add_entities(
+                [
+                    Control4Light(
+                        hass,
+                        entry,
+                        item_coordinator,
+                        item_name,
+                        item_id,
+                        item_device_name,
+                        item_manufacturer,
+                        item_model,
+                        item_parent_id,
+                        item_is_dimmer,
+                        light_transition_time,
+                        light_cold_start_transition_time,
+                    )
+                ],
+                True,
+            )
+
+
+class Control4Light(Control4Entity, LightEntity):
+    """Control4 light entity."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+        coordinator: DataUpdateCoordinator,
+        name: str,
+        idx: int,
+        device_name: str,
+        device_manufacturer: str,
+        device_model: str,
+        device_id: int,
+        is_dimmer: bool,
+        light_transition_time: int,
+        light_cold_start_transition_time: int,
+    ):
+        """Initialize Control4 light entity."""
+        super().__init__(
+            hass,
+            entry,
+            coordinator,
+            name,
+            idx,
+            device_name,
+            device_manufacturer,
+            device_model,
+            device_id,
+        )
+        self._is_dimmer = is_dimmer
+        self._transition_time = light_transition_time
+        self._cold_start_transition_time = light_cold_start_transition_time
+        self._c4_light = C4Light(self.director, idx)
```

Comment:
Please move side effects out of `__init__`

Myabe create this in `async def async_added_to_hass`